memory-game.js

```diff
@@ -152,36 +159,101 @@ function createCard(type, data, imgBackSrc) {
   card.appendChild(frontContent);
   card.appendChild(imgBack);
   cardContainer.appendChild(card);
-
   card.addEventListener("click", function () {
+    if (activeCards.length >= 2 || card.classList.contains("isFlipped")) {
+      return;
+    }
     card.classList.toggle("isFlipped");
+    document.getElementById("counter").innerHTML = ++numberOfMoves;
+    console.log(numberOfMoves);
+    activeCards.push(card);
+
+    if (activeCards.length === 1 && !timerStarted) {
+      startTimer();
+    }
+    if (activeCards.length === 2) {
+      setTimeout(() => {
+        activeCards.forEach((card) => {
+          card.classList.remove("isFlipped");
+        });
+        activeCards = [];
+      }, 2000);
+    }
   });
   return cardContainer;
 }
 
-function populateGrid(category, grid) {
-  grid.innerHTML = "";
-  let combinedArray = [];
-  if (category === "animals") {
-    combinedArray = images.concat(
-      animals.map((animal) => ({ type: "word", data: animal }))
-    );
-  } else if (category === "weather") {
-    combinedArray = imagesWeather.concat(
-      weather.map((weather) => ({
-        type: "word",
-        data: weather,
-      }))
-    );
-  }
-  shuffle(combinedArray);
-  combinedArray.forEach((item) => {
-    if (item.url) {
-      const card = createCard("image", item.url, item["back-cover"]);
-      grid.appendChild(card);
-    } else if (item.data) {
-      const card = createCard("word", item.data, "./assets/back.jpg");
-      grid.appendChild(card);
-    }
+document.addEventListener("DOMContentLoaded", () => {
+  const overlay = document.querySelector(".overlay-text");
+  overlay.classList.add("visible");
+
+  const grid = document.getElementById("grid-container");
+  grid.classList.add("disabled");
+  const startGameBtn = document.getElementById("start-game");
+  const restartGameBtn = document.getElementById("restart-game");
+
+  startGameBtn.addEventListener("click", () => {
+    overlay.classList.remove("visible");
+    grid.classList.remove("disabled");
   });
+  restartGameBtn.addEventListener("click", (e) => {
+    const overlays = document.querySelectorAll(".overlay-text");
+    overlays.forEach((item) => item.classList.remove("visible"));
+    grid.classList.add("disabled");
+    resetTimer();
+    resetMoves();
+    populateGrid(document.getElementById("category").value, grid);
+
+    grid.classList.remove("disabled");
+  });
+
+  document.getElementById("category").addEventListener("change", (e) => {
+    const selectedCategory = e.target.value;
+    console.log(selectedCategory);
```

Comment:
In general, it is recommended not to leave console.logs in the code that is committed to the repository.